python_modules/dagstermill/dagstermill/__init__.py

```diff
@@ -156,33 +179,54 @@ def populate_context(
         marshal_dir,
         environment_dict,
         output_log_path,
+        input_name_type_dict,
+        output_name_type_dict,
     ):
         check.dict_param(environment_dict, 'environment_dict')
         self.populated_by_papermill = True
-        check.invariant(
-            self.repository_def != None,
-            desc='When running Dagstermill notebook in pipeline, '
-            'must register a repository within notebook by calling '
-            '"dm.register_repository(repository_def)"',
-        )
-        self.pipeline_def = self.repository_def.get_pipeline(pipeline_def_name)
-        check.invariant(self.pipeline_def.has_solid_def(solid_def_name))
-        self.solid_def = self.pipeline_def.solid_def_named(solid_def_name)
-
+        self.solid_def_name = solid_def_name
         self.marshal_dir = marshal_dir
-        loggers = None
-        if output_log_path != 0:
-            event_logger = construct_json_event_logger(output_log_path)
-            loggers = [event_logger]
-        # do not include event_callback in ExecutionMetadata,
-        # since that'll be taken care of by side-channel established by event_logger
-        run_config = RunConfig(run_id, loggers=loggers)
-        # See block comment above referencing this issue
-        # See https://github.com/dagster-io/dagster/issues/796
-        with yield_pipeline_execution_context(
-            self.pipeline_def, environment_dict, run_config
-        ) as pipeline_context:
-            self.context = DagstermillInNotebookExecutionContext(pipeline_context)
+
+        if self.repository_def is None:
+            self.pipeline_def = PipelineDefinition([], name='Dummy Pipeline (No Repo Registration)')
```

Comment:
@schrockn is it just me or might it be nice to have like a dummy pipeline factory :)